image.md

```diff
@@ -0,0 +1,140 @@
+
+1. Whiteout list.
+An image MAY choose to include a list of files that are to be considered to be deleted from the root filesystem even though they are included within the image's root filesystem directory.
+It is an implementation detail to decide how this whiteout list is processed but an implementation MUST ensure that any container that is created from this image will behave the same as if those files were not in the root filesystem at all.
+The whiteout list MUST be in a file called `whiteout.json`.
+See the [Whiteout List](#whiteout-list) section for more details.
+
+1. Extension files.
+An image MAY choose to include additional files in the image.
+
+1. The ID of the image.
+This REQUIRED information MUST be in a file called `ID`.
+See the [Image ID](#image-id) section for more details.
+
+## Serialization of an Image
+
+An image MUST be a `tar` of the pieces of data that make up an image as specified in the [Image Contents](#image-contents) section.
+The order in which the data is serialized in the `tar` MUST be the same as the order specified in that section.
+The serialization of the "dependent images" MUST be in the same order in which those dependent images are listed in the `requires.json` file.
+The serialization of the  "extension files" MUST be in case-sensitive alphabetical order.
+
+Each file or directory specified in the [Image Contents](#image-contents) section MUST appear at the root of the `tar` without a leading `/` in their names.
+
+For example, a listing of the `tar` of an image might look like this:
+```
+config.json
+rootfs/
+    bin/myapp
+requires.json
+sha512-1d62d181e9c1322d56ccd3a29d05018399147a16188dbd861c0279ad0dd7e14c/
+    config.json
+	rootfs/
+	    bin/runit
+	whiteout.json
+	ID
+sha512-291e2a171ef9bef8a838c59406d9b0aeb6f2f0ebe5173415205733d3d18b8e03/
+    config.json
+	rootfs/
+	    bin/monitor
+	whiteout.json
+	ID
+whiteout.json
+ID
+```
+
+## Dependent Images
+
+If an image has dependent images then it MUST include a `requires.json` file that references those images.
+The `requires.json` file MUST be a json encoded array matching this format:
+```
+{
+  "requires": [
+    {
+      "ID": "..."
+    },
+	{
+	  "ID": "..."
+	}, ...
+  ]
+}
+```
+
+The order of the images, `ID`s, in the array MUST be in the order in which the images are to be "layed down" into the filsystem bundle that is generated from this image.
+In other words, the first image in the array is written to the filesystem first, and subsequent images are overlayed on top - potentially overlaying an ealier image's files.
+
+QUESTION: what about deleted files? Do we want to mandate a whiteout scheme?
+
+## Image ID
+
+An image's `ID` MUST be a sha512 hash of image's contents in the order specified in the [Image Contents](#image-contents) section.
+
+An image's `ID` MUST be a string of the form `sha512-<hash>`.
+
+QUESTION: Do we really want to mandate sha512?
```

Comment:
It is a solid default. For a majority of x86{,_64} chips, it is faster than sha256. Though for architectures like ARM, this could be a hit. 
Here is my laptop. Both are speedy, but sha512 is faster:
```bash
$ dd if=/dev/urandom of=rand.img bs=1M count=64
64+0 records in
64+0 records out
67108864 bytes (67 MB) copied, 9.47004 s, 7.1 MB/s
$ time sha256sum rand.img 
2d4e6875974352af656630e8ed76ee036847888d69c507f73809d672c58f439d  rand.img

real	0m1.068s
user	0m1.026s
sys	0m0.041s

$ time sha512sum rand.img 
ba914581b051a017d9f7fc3169bd6036bfa24666eb692d891107ce237d60a091919f0cbce0408221c452bd0217ee1388a72ca33a2b65e6bb791714217f85a4d3  rand.img

real	0m0.692s
user	0m0.654s
sys	0m0.036s
```

Here is a raspberry pi. The sha512 is an order of magnitude slower:
```bash
$ dd if=/dev/urandom of=rand.img bs=1M count=64
64+0 records in
64+0 records out
67108864 bytes (67 MB) copied, 54.8794 s, 1.2 MB/s
$ time sha256sum rand.img
d286218629fccb13c9c1568b6ddb9f2fc5452299a4ee307e3cff353836d8901a  rand.img

real    0m5.552s
user    0m4.850s
sys     0m0.440s
$ time sha512sum rand.img
8dcaddd1337cc9dcef273b4a53243563a619747214256a7c6285e6e7a94c671d06121ba08c87ae8c0a33bde4eb142d80413782913f79d9fd2686799174be1061  rand.img

real    1m38.730s
user    1m33.010s
sys     0m1.120s
```

But like the pattern of `<algorithm>-<digest>` could allow folks to choose alternates that suite their needs better, or once there is the next great algorithm to be switched to.